src/pudl/output/ferc1.py

```diff
@@ -1,13 +1,41 @@
 """Functions for pulling FERC Form 1 data out of the PUDL DB."""
 import numpy as np
 import pandas as pd
+import sqlalchemy as sa
 
 import pudl
 from pudl.metadata.fields import apply_pudl_dtypes
 
 logger = pudl.logging_helpers.get_logger(__name__)
 
 
+def read_table_with_start_end_dates(
+    table_name: str,
+    start_date: pd.Timestamp,
+    end_date: pd.Timestamp,
+    pudl_engine: sa.engine.Engine,
+) -> pd.DataFrame:
+    """Read a FERC1 table with a start and end date restriction.
+
+    Note: This is a FERC1 function instead of a general function because FERC1 uses
+    ``report_year`` instead of the more standard ``report_date``.
+
+    Args:
+        table_name: Name of FERC1 table.
+        start_date: Beginning date for data to pull from the PUDL DB.
+        end_date: End date for data to pull from the PUDL DB.
+        pudl_engine: : A connection engine for the PUDL DB.
+    """
+    pt = pudl.output.pudltabl.get_table_meta(pudl_engine)
+    table = pt[table_name]
+    table_select = (
+        sa.sql.select(table)
+        .where(table.columns.report_year >= f"{start_date.year}")
+        .where(table.columns.report_year <= f"{end_date.year}")
+    )
+    return pd.read_sql(table_select, pudl_engine)
```

Comment:
I ended up moving everything including the metadata generation and actually reading the table into this lil function. It felt cleaner to do the generation of `table` here all in one place as opposed to in all the functions. I could go either way with the actual `pd.read_sql` step being included in here.